Change logging documentation (#91)
yiisoft/docs: guide/en/runtime/logging.md

Yii comes with the following built-in log targets. Please refer to the API documentation about these classes to 
learn how to configure and use them. 

> * [[\Yiisoft\Log\PsrTarget]]: passes log messages to another PSR-3 compatible logger.
> * [[\Yiisoft\Log\StreamTarget]]: writes log messages in specified output stream.
> * [[\Yiisoft\Log\Target\Db\DbTarget]]: saves log messages in database.

Reviewer: ```suggestion
* [[\Yiisoft\Log\PsrTarget]]: passes log messages to another PSR-3 compatible logger.
* [[\Yiisoft\Log\StreamTarget]]: writes log messages into specified output stream.
* [[\Yiisoft\Log\Target\Db\DbTarget]]: saves log messages in database.
```

### Message Formatting <span id="message-formatting"></span>

Log targets export the filtered log messages in a certain format.
For example, if you install a log target of the class [[\Yiisoft\Log\Target\File\FileTarget]],
then by default the log messages will be formatted as follows:

```

> Timestamp Prifix[Level][Category] Message Context

Reviewer: ```suggestion
Timestamp Prefix[Level][Category] Message Context
```

Reviewer: A real example was handy here. I think it's a good idea to provide it as well.

```

You may customize this format by calling [[\Yiisoft\Log\Target::setFormat()|setFormat()]] method,

> which takes a PHP callable returning a customized message format.

Reviewer: ```suggestion
which takes a PHP callable returning a custom formatted message.
```

and [[\Yiisoft\Log\Target::disable()|disable()]] methods.
You may do so via the log target configuration or by the following PHP statement in your code:

```php
$fileTarget = new \Yiisoft\Log\Target\File\FileTarget('/path/to/app.log');
$logger = new \Yiisoft\Log\Logger([$fileTarget, /*Other targets*/]);

foreach ($logger->getTargets() as $target) {
    if ($target instanceof \Yiisoft\Log\Target\File\FileTarget) {
        $target->disable();
    }
}
```

To check whether the log target is enabled, call the `isEnabled()` method.
You also may pass a callable to [[\Yiisoft\Log\Target::setEnabled()|setEnabled()]]
to define a dynamic condition for whether the log target should be enabled or not.

### Creating New Targets <span id="new-targets"></span>

Creating a new log target class is very simple. You mainly need to implement the [[\Yii\Log\Target::export()]]
abstract method that sends all accumulated log messages to a designated medium.

The following protected methods will also be available for child targets:

- `getMessages` - Gets a list of log messages ([[\Yii\Log\Message]] instances).

> - `getFormattedMessages` - Gets a list of formatted string log messages.

Reviewer: ```suggestion
- `getFormattedMessages` - Get a list of log messages formatted as strings.
```

> - `formatMessages` - Gets formatted all log messages for display as a string.

Reviewer: ```suggestion
- `formatMessages` - Get all log messages formatted as a string.
```

> - `getCommonContext` - Gets an array with common context data in the `key => value` format.

Reviewer: ```suggestion
- `getCommonContext` - Get an array with common context data in the `key => value` format.
```

> For more details, you may refer to any of the log target classes included in the Yii release.

Reviewer: ```suggestion
For more details, you may refer to any of the log target classes included in the package.
```
